ws-backend/src/admin/admin.controller.ts

```diff
@@ -0,0 +1,170 @@
+import {
+  Body,
+  Controller,
+  Get,
+  NotFoundException,
+  Param,
+  Post,
+} from '@nestjs/common';
+import { AdminService } from './admin.service';
+
+@Controller('admin')
+export class AdminController {
+  constructor(private readonly adminService: AdminService) {}
+
+  /*QUEUES*/
+  @Get('/get-all-queues')
+  async getAllQueues(): Promise<object> {
+    const res = await this.adminService.getAllQueues();
+    if (null === res) {
+      throw new NotFoundException({
+        code: 'NOT_FOUND',
+        message: `not found any queues`,
+      });
+    }
+    return res;
+  }
+
+  @Get('/get-queue/:idQueue')
+  async getQueue(@Param('idQueue') idQueue: number): Promise<object> {
+    const res = await this.adminService.getQueue(Number(idQueue));
+    if (null === res) {
+      throw new NotFoundException({
+        code: 'NOT_FOUND',
+        message: `not found any queue with id = ${idQueue}`,
+      });
+    }
+    return res;
+  }
+
+  @Post('/edit-queue/:idQueue')
+  async editQueue(
+    @Param('idQueue') idQueue: number,
+    @Body() payload: object,
+  ): Promise<object> {
+    const res = await this.adminService.editQueue(idQueue, payload);
+    return res;
+  }
+
+  @Post('/add-queue')
+  async addQueue(@Body() payload: object): Promise<object> {
+    const res = await this.adminService.addQueue(payload);
+    return res;
+  }
```

Comment:
No need to wrap the function with `async` keyword. Just return whatever `addQueue` returns if there is no operation being performed on the returned object.
```suggestion
  addQueue(@Body() payload: object): Promise<object> {
    return this.adminService.addQueue(payload);
  }
```